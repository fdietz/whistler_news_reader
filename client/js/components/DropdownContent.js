import React, { PropTypes } from 'react';
import classNames from 'classnames';

const DropdownContent = ({ children, className, style, active }) => {
  const cls = classNames('dropdown__content', className, {
    hidden: !active,
  });

  return (
    <div className={cls} style={style}>
      {children}
    </div>
  );
};

DropdownContent.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  style: PropTypes.object,
  active: PropTypes.bool,
};

export default DropdownContent;
