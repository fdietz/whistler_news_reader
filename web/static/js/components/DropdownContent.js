import React, { PropTypes, Component } from "react";
import classNames from "classnames";

class DropdownContent extends Component {

  static propTypes = {
    children: PropTypes.node,
    className: PropTypes.string,
    style: PropTypes.object,
    active: PropTypes.bool
  };

  render() {
    const { children, className, style, active } = this.props;

    const cls = classNames("dropdown__content", className, {
      hidden: !active
    });

    return (
      <div className={cls} style={style}>
        {children}
      </div>
    );
  }
}

export default DropdownContent;
