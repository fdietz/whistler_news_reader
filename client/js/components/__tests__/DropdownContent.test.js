import renderer from 'react-test-renderer';
import React from 'react';
import DropdownContent from '../DropdownContent';

describe('DropdownContent', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<DropdownContent />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
