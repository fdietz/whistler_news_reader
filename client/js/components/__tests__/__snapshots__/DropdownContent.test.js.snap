exports[`DropdownContent renders correctly 1`] = `
<div
  className="dropdown__content hidden"
  onClick={undefined}
  style={undefined} />
`;
